.github/helper/semgrep_rules/ux.yml

```diff
@@ -7,6 +7,14 @@ rules:
   - patterns:
       - pattern: frappe.throw("...", ...)
       - pattern-not: frappe.throw(_("..."), ...)
+  - patterns:
+      - pattern: |
+          {..., "label": "...", ...}
+      - pattern-not: |
+          {..., "label": _("..."), ...}
+  - patterns:
+      - pattern: dict(..., label="...", ...)
+      - pattern-not: dict(..., label=_("..."), ...)
```

Comment:
One minor change to avoid false +ves: Target this to `**/report/` glob pattern only, you'll have to duplicate the rule for this.
example:
https://github.com/frappe/erpnext/blob/a60a3d1c8b6a3e3f2eeb0c495ebb39a403382441/.github/helper/semgrep_rules/frappe_correctness.yml#L100-L102